flax/linen/attention.py

```diff
@@ -142,7 +144,7 @@ class MultiHeadDotProductAttention(Module):
   out_features: Optional[int] = None
   broadcast_dropout: bool = True
   dropout_rate: float = 0.
-  deterministic: bool = False
+  deterministic: Optional[bool] = None
```

Comment:
Should this be a separate CHANGELOG entry? i.e. "Remove deterministic default for ..."